### DISCUSSION TITLE
Please disable pretty_exception_enable by default

### DISCUSSION THREAD

**Discussion by LucaBonfiglioli** [fastapi/typer#654] [Questions] [answered]
### First Check

- [x] I added a very descriptive title here.
- [x] I used the GitHub search to find a similar question and didn't find it.
- [x] I searched the Typer documentation, with the integrated search.
- [x] I already searched in Google "How to X in Typer" and didn't find any information.
- [x] I already read and followed all the tutorial in the docs and didn't find an answer.
- [x] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
import typer
voglio_morire = typer.Typer(name="voglio_morire", pretty_exceptions_enable=False)
```

### Description

Exception pretty printing is harmful and completely useless in basically every situation I can think of. Plus, it's not safe to turn it off by simply passing:

```
pretty_exceptions_enable=False
```

Because this will add an implicit dependency to your code: in order to do this you need to have typer>=0.7, otherwise, an error will be raised. The easiest solution is to manually check the typer version, and then disabling the option if it is >= 0.7

Pretty print exception only works in toy examples when the stack has a depth of 1 to 3, unless you like to have your terminal completely cluttered with magical rainbow stuff, which more often than not exceeds the tty length, you should not use this in a real-world scenario. Please disable it by default.

### Operating System

Linux, Windows, macOS, Other

### Typer Version

>=0.7

### Python Version

Any

**Comment by pamelafox** [comment]
I am finding myself turning it off as well, as the tracebacks have such depth as to be overwhelming. I'm not sure if there's a bug that's resulting in really high depth tracebacks or if that's to be accepted (even in "short" mode), but I'm going to disable entirely.
I'm curious what the experience is like across developers as a whole with the tracebacks, across developers of new/int/advanced experience.

**Comment by dyollb** [comment]
I also wanted to disable `rich` exception handling, since its so annoying to always scroll and carefully search for the source of the exception (see example below).

In the documentation, it says this feature is only active [if you install the **optional** dependency rich](https://typer.tiangolo.com/tutorial/exceptions/#exceptions-with-rich), however if I do `pip install typer` rich is also installed. If I remove rich I get an error.
![image](https://github.com/tiangolo/typer/assets/12702862/e6ad4f44-d8ab-40eb-80d7-01af7ef53f29)


### Example pretty exception
![rich_exceptions](https://github.com/tiangolo/typer/assets/12702862/3e482ef8-3823-4213-b6f8-68931a7d489d)

**Reply by BeCracked** [reply]
I also find myself disabling pretty exceptions when developing. This is mainly because my IDE (PyCharm atm) allows me to click the lines in the traceback with the plain Python tracebacks. This is not the case with the rich tracebacks. Not really rich's fault, but it is how it is.

**Reply by albalamia** [reply, marked as answer]
You can install the minimal version which does not come with Rich
`pip install typer-slim`

**Reply by dyollb** [reply]
Thanks @albalamia i didn't see this in the documentation! This solves the issue for me

**Reply by tiangolo** [reply]
Thanks for the screenshot @dyollb! This was very illustrative of the problem. I decided to change the default because of this. :nerd_face:
